### ISSUE TITLE
[Feature Request]: Add support for Milvus Vector Database

### ISSUE THREAD

**Issue by AbhishekRP2002** [truefoundry/cognita#415]
Milvus is a high-performant, scalable open-source vector database which is adopted by multiple industry leaders ( Salesforce, Nvidia, Walmart to name a few . ref: https://milvus.io/use-cases ) to power up their AI applications.

Since, it's used in enterprise level, so integrating it with cognita can be beneficial for folks working around data augmented LLM agentic applications.

Scope:

- Milvus supports 3 deployment modes : Milvus Lite, Milvus Standalone and Milvus Distributed.
- We can start with integrating Milvus Lite, a lightweight version of Milvus via their python client library. Post successful integration of Milvus Lite, we can support `Milvus Standalone and Distributed` since all the deployment modes of Milvus share the same API, so the client side code doesn’t need to change much while migrating from one deployment mode to another.
ref: https://milvus.io/docs/overview.md

**Comment by AbhishekRP2002** [comment]
I would love to take this up. 
cc: @chiragjn

**Comment by chiragjn** [comment]
Hey @AbhishekRP2002, sure we would love to have basic Milvus integration. Please feel free to contribute and ask for help

**Comment by AbhishekRP2002** [comment]
Yes. I've started working on it. Aiming to make a PR by the end of this week.
